### PR TITLE
docs(tracing): adds DD_TRACE_PARTIAL_FLUSH_MIN_SPANS to configurations

### PR DESCRIPTION
**Pull request by mabdinur**
This PR ensures ``DD_TRACE_PARTIAL_FLUSH_ENABLED`` is documented. 

Partial flushing is an implementation detail of the tracer so I don't think we need to document this configuration in https://github.com/DataDog/documentation (this justification is pretty hand wavy, feel free to push back).

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: docs/configuration.rst [hunk 1]
```diff
@@ -368,6 +368,11 @@ The following environment variables for the tracer are supported:
      default: True
      description: Prevents large payloads being sent to APM.
 
+    DD_TRACE_PARTIAL_FLUSH_MIN_SPANS:
+     type: String
```
Comment (ZStriker19) [inline docs/configuration.rst:372]: shouldn't type be `int` ?

Comment (mabdinur) [inline docs/configuration.rst:372]:
```suggestion
     type: Integer
```

File: docs/configuration.rst [hunk 2]
```diff
@@ -368,6 +368,11 @@ The following environment variables for the tracer are supported:
...
+     default: 500
+     description: Maximum number of spans sent per trace per payload when ``DD_TRACE_PARTIAL_FLUSH_ENABLED=True``
```
Comment (mabdinur) [inline docs/configuration.rst:374]:
```suggestion
     description: Maximum number of spans sent per trace per payload when ``DD_TRACE_PARTIAL_FLUSH_ENABLED=True``.
```

**Comment by ZStriker19** [comment]
Should the title be for ` DD_TRACE_PARTIAL_FLUSH_MIN_SPANS` ? I think it's ok to not add this to Datadog documentation, as this seems to follow with what other libraries have done.

**Review by ZStriker19 — DISMISSED** [review]

File: docs/configuration.rst [hunk 3]
```diff
@@ -368,6 +368,11 @@ The following environment variables for the tracer are supported:
      default: True
      description: Prevents large payloads being sent to APM.
 
+    DD_TRACE_PARTIAL_FLUSH_MIN_SPANS:
```
Comment (mabdinur) [inline docs/configuration.rst:371]:
```suggestion
   DD_TRACE_PARTIAL_FLUSH_MIN_SPANS:
```

**Comment by majorgreys** [comment]
What is the motivation to add this to our documentation? While we document `DD_TRACE_PARTIAL_FLUSH_ENABLED`, we do communicate when/how to use this feature. 

I also think we should be cautious in documenting configuration options if that impacts our versioning policy. For example, if we were to rewrite the partial flushing with different configuration options, we would have to continue supporting `DD_TRACE_PARTIAL_FLUSH_MIN_SPANS` if it was documented.

**Comment by mabdinur** [comment]
> What is the motivation to add this to our documentation? While we document `DD_TRACE_PARTIAL_FLUSH_ENABLED`, we do communicate when/how to use this feature.
> 
> I also think we should be cautious in documenting configuration options if that impacts our versioning policy. For example, if we were to rewrite the partial flushing with different configuration options, we would have to continue supporting `DD_TRACE_PARTIAL_FLUSH_MIN_SPANS` if it was documented.


A user reached out to our support team about this configuration. They wanted to set this configuration to resolve an issue but were hesitant to set it since it was undocumented. In the past we encouraged users to set this configuration so I think it should be documented. 

I interpreted our version policy for configurations as:
-  All configurations prefixed with an underscore (ex: ``_DD``) are internal. 
- All configurations without the `_` prefix is "public" and can/should be documented.

**Review by ZStriker19 — APPROVED** [review]

**Pull request merged**